### ISSUE TITLE
autohide_bookmarks.toolbar.css issue 

### ISSUE THREAD

**Issue by kenarsene** [MrOtherGuy/firefox-csshacks#257]
New firefox update v110 makes the bookmarks bar transparent, any fix?

![Screenshot 2023-02-15 045831](https://user-images.githubusercontent.com/58550960/218862004-37b506cd-9e9e-4bf7-a5ff-6e07b023a1a9.png)

**Comment by kenarsene** [comment]
The problem is `background-color: transparent !important;` deleted that and it's fine now.

**Comment by MrOtherGuy** [comment]
Yeah, so this happens because Firefox removed --toolbar-bgimage cuatom property that was used in bunch of toolbars as background. So now that it doesn't exist the `background-image` we use on bookmarks toolbar computes to "none" thus you see it transparent.

We can't really do what you did and just remove that background-color: transparent rule because that's only going to work if one uses a Firefox theme with opaque, solid color toolbars. 

In b906186 I added a temporary workaround by providing a fallback for case where that variable is missing which should make things work. But it looks like that variable is indeed completely removed from Firefox codebase and we can simplify the background-image related properties a bit. But I'll do that later.

It might also be worth to keep this hotfix since it should also work on Firefox ESR 102 and then only simplify the styling when next ESR releases.

**Comment by onmyouji** [comment]
The latest Firefox 150 makes the bookmark bar transparent again. Anyone know how to fix it?

**Comment by MrOtherGuy** [comment]
Well for the record, Firefox 150 isn't available yet on my system, but on Nightly 152 the bookmarks toolbar is still visible and not transparent. Please test if the transparent look persists when you only use autohide_bookmarks_toolbar.css and no other styles.

That being said, based on that old comment of mine above, there might be stuff we could do to simplify bookmarks_toolbar.css

**Comment by onmyouji** [comment]
Sorry I forgot to mention, I use the vertical tabs from Firefox settings (tabs on the left side, bookmark on the top).

<img width="860" height="96" alt="Image" src="https://github.com/user-attachments/assets/8687b5ec-8213-4972-9865-6468a345e68f" /><br/>

Turns out this problem only happens when using vertical tabs. If I switch to normal horizontal tabs from Firefox settings, the bookmark toolbar is visible (not transparent).

<img width="721" height="198" alt="Image" src="https://github.com/user-attachments/assets/b6a96f76-973d-4060-a6c9-fb8c9e034754" /><br/>

But this problem only starts happening after updating to 150. Before 150, the bookmark toolbar is visible on both vertical and horizontal tab layout.

**Comment by MrOtherGuy** [comment]
Yeah okay, I know what's going on now. But I'm not sure how I want to fix this, needs a bit more thought.

In the mean time you can just slap a quick `#PersonalToolbar#PersonalToolbar{ background-color: var(--toolbox-bgcolor) !important; }` after other rules and it should work.

**Comment by onmyouji** [comment]
Thanks!
